### PR TITLE
feat(ch04s06): Provide frontend for books api

### PR DESCRIPTION
**Pull request by m-czernek**

### CODE REVIEW

File: library/books-frontend/src/components/BookCard.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+                <p>Year of publishing: {year}</p>
+                <p>Copies available: {copies}</p>
+              </TextContent>
+            </CardBody>
+          </CardExpandableContent>
+        </Card>
+      </>
+    );
+  }
+}
```
Comment (gjbianco) [inline library/books-frontend/src/components/BookCard.js]: With the addition of `useState`, I can't think of a reason to use class-based components over function-based components anymore 🙂 

```suggestion
export default function BookCard(props) {
  const { year, copies, title } = props;

  const [isExpanded, setIsExpanded] = useState(false);
  
  return (
    <>
      <Card style={{margin: "20px"}} isExpanded={isExpanded} >
        <CardHeader onExpand={() => setIsExpanded(!isExpanded)}>
          { title }
         </CardHeader>
         <CardExpandableContent>
           <CardBody>
             <TextContent>
               {/* <p>Author: {props.authorName}</p> */}
               <p>Year of publishing: {year}</p>
               <p>Copies available: {copies}</p>
             </TextContent>
           </CardBody>
         </CardExpandableContent>
       </Card>
       </>
  );
}
```

File: library/books-frontend/src/App.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
+import React, { useEffect, useState } from "react";
+import { Brand, Page, PageSection, PageSectionVariants, Text, TextContent, Bullseye, PageHeader } from "@patternfly/react-core";
+import getBooks from "./services/bookService";
+import BookCard from "./components/BookCard";
+import imgBrand from "./images/training_white.png";
+import { useKeycloak } from "@react-keycloak/web";
+
+export default function Dashboard() {
+  const [bookData, setBookData] = useState([]);
```
Comment (gjbianco) [inline library/books-frontend/src/App.js:9]: Side note: this is a good example of why I like using TS with React.

As is, the type of `bookData` is `any[]`. With TS, you could write this line as:

```tsx
const [bookData, setBookData] = useState<Book[]>([]);
```

And then of course you'd have an interface for `Book` somewhere.

That way, anytime you reference an item in `bookData`, you get autocompletion and (compile-time) type safety checks.

File: library/books-frontend/src/App.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  const { keycloak, initialized } = useKeycloak();
```
Comment (gjbianco) [inline library/books-frontend/src/App.js:10]: I haven't used `@react-keycloak/web` before. This is super cool that you can do this now!

**Review by gjbianco — APPROVED** [review]
Very clean! Just one minor code suggestion.

Also, I recommend removing `yarn.lock` since you already have `package-lock.json` and don't list `yarn` as a dependency.

NB: I did not actually test the functionality btw. I just reviewed the code. To my eyes, everything looks like it should work fine and I assume you've run it lol.

**Review by alxlenc — APPROVED** [review]
Since I have tested the front-end app, I can approve the functionality part.

File: library/books-frontend/src/components/BookCard.js (hunk 1 above)
Comment (m-czernek) [reply library/books-frontend/src/components/BookCard.js]: +1, the original reason was for having multiple methods, but I cut them in the end, and you're right that the `onExpand` method was easy to bind by using an arrow function. Applying.

File: library/books-frontend/src/App.js (hunk 2 above)
Comment (m-czernek) [reply library/books-frontend/src/App.js:9]: I'll keep TS in mind; definitely want to change this to TS but will do so later if time allows.

**Comment by m-czernek** [comment]
Thank you folks! I have changed one class to functional component, and will keep TS in mind. Merging for now.

**Pull request merged**